Modules/Module_Import/Import/xASL_imp_DCM2NII_Subject_StartConversion.m

```diff
@@ -58,12 +58,27 @@
             %% Start the conversion. Note that the dicom filter is only in effect when a directory is specified as input.
             try
                 % First we try to see if there are deeper layers
-                [fPath, ~, fExt] = xASL_fileparts(scanpath);
-                if isempty(fExt) % run this part only for folders
-                    filepathsNonRecursive = xASL_adm_GetFileList(fPath, x.modules.import.imPar.dcmExtFilter, 'FPList');
-                    filepathsRecursive = xASL_adm_GetFileList(fPath, x.modules.import.imPar.dcmExtFilter, 'FPListRec');
-                    if isempty(filepathsNonRecursive) && ~isempty(filepathsRecursive)
-                        scanpath = fileparts(filepathsRecursive{1});
+                if xASL_exist(scanpath, 'dir') == 7 % run this part only for folders
+                    filepathsNonRecursive = xASL_adm_GetFileList(scanpath, x.modules.import.imPar.dcmExtFilter, 'FPList'); % List files directly in scanpath
+                    filepathsRecursive = xASL_adm_GetFileList(scanpath, x.modules.import.imPar.dcmExtFilter, 'FPListRec'); % List all files including subdirectories
+                    if isempty(filepathsNonRecursive) && ~isempty(filepathsRecursive) 
+						% There are no files in the directory but there are files in the subdirectories
+						for iFile = 1:length(filepathsRecursive)
+							% Remove the filename and keep only the path
+							filepathsRecursive{iFile} = xASL_fileparts(filepathsRecursive{iFile});
+						end
+						% Find the unique paths
+						filepathsRecursiveUnique = unique(filepathsRecursive);
+						if isempty(filepathsRecursiveUnique)
+							% No files found recursively -> error
+							error(['No files found in ' scanpath]);
+						elseif length(filepathsRecursiveUnique) > 1
+							% Files at multiple directory levels -> error
+							error(['Multiple directory levels with files were found in ' scanpath]);
+						else
+							% Only a single directory level with files found - use that one
```

Comment:
You're saying "single directory level" but it can be that you find a "single directory level" but on this same "level" you have multiple directories. Just to have a correct error message and comment, for the user to understand.

E.g.:

/dicomFolderTokenSpecified/ExtraDirectoryLevel/ControlSiemens/dicoms.dcm
/dicomFolderTokenSpecified/ExtraDirectoryLevel/LabelSiemens/dicoms.dcm

so the error message should be:

error(['Multiple subdirectories with (DICOM) files were found in ' scanpath]);

and the comments should be like:
% Only a single subdirectory with files found - use that one

so call them "subdirectories" instead of "directory levels" because you don't discriminate "horizontal vs vertical" levels if you understand what I mean

Reply:
Thanks. It was indeed tricky to nicely describe this.

Code changed accordingly.